### ISSUE TITLE
Release 2.0.0

### ISSUE THREAD

**Issue by kalicki1** [privacy-tech-lab/gpc-optmeowt#197]
Since we're getting ready to push a release, I figured we could open an issue to keep track of outstanding issues and have a place to discuss release procedures. 

I think what remains to be finalized are: 

- [x] import/export settings page functionality - issue #193 
- [x] Tentatively rename files for finalized commit - issue #172 
- [x] Update developer names in source code - issue #160 
- [x] dark mode - issue #183
- [x] Fix header signal not sent in Chrome - issue #200
- [x] Fix domain deletion and re-adding bug - issue #202 
- [x] GPC signal is sent despite being disabled on Firefox - issue #203
- [x] Update wiki - issue #198 

We should also make sure our readme and other documentation are up to date (related issue #188). 

Then we should do one final performance test, review all the code, and get ready for release. I will review Firefox's release requirements for when we are ready to go. We should also remember to release for the Chrome store.

**Comment by kalicki1** [comment]
We should note that this release is also significant because it fixes the GPC DOM injector according to issue #195.

**Comment by kalicki1** [comment]
I'm wrapping up double checking the code and testing for correct logic, and opened issue #199 to help resolve some quirks I found. I should be able to do this fairly quickly .

**Comment by kalicki1** [comment]
I just pushed a few fixes regarding several issues I managed to catch. (1) I pushed a change fixing the logic of how the extension's mode is handled (i.e., is OptMeowt enabled/disabled/domainlisted) (Regarding this, I think we should add a 3 way switch or something similar to the popup in the future). (2) I also pushed a change that fixes rendering issues on the popup when a given tab's domain is `null` or `undefined`. 

I'm fairly confident that's most of what there was left to fix. I will start getting ready to package everything together tomorrow morning, and in the meantime, I can do one last test run to make sure I didn't miss anything else

@ everyone, let me know if you all find anything else that needs to be fixed while testing if you happen to check it out yourselves. I'm excited to see were so close!

**Comment by SebastianZimmeck** [comment]
Indeed, 2.0.0 ... :)

**Comment by kalicki1** [comment]
I just pushed some notes regarding issue #183 and issue #203, I believe the two outstanding issues. At this point in time I think we are basically done and can move forward with pushing our release in the morning.

**Comment by kalicki1** [comment]
I'm going to go ahead and push a new release for v2.0.0 and begin uploading our updated extension to the Firefox Add-Ons store. 

I believe the requirements are that if we do have built or obfuscated code, we need a link and instructions on how to build in some public source code, which we do. I will go ahead with making this happen.

**Comment by SebastianZimmeck** [comment]
Great! I will submit the new release to the Chrome Web Store.

**Comment by SebastianZimmeck** [comment]
I ran into some issues on the Chrome Web Store. Our 2.0.0 release was rejected. I contacted the support for the exact reason so that we can fix it.

**Comment by SebastianZimmeck** [comment]
OK, I resubmitted on the Chrome Web Store. This time it should pass. I just submitted the wrong files the last time. I totally forgot that we are now using Webpack etc. and instead of submitting the `dist` files for `chrome` I submitted the `src` files. So, this should sort itself out soon.

**Comment by SebastianZimmeck** [comment]
@kalicki1, it would be good to decide in our meeting tomorrow on the status of the [readme in the background folder](https://github.com/privacy-tech-lab/gpc-optmeowt/blob/main/src/background/readme.md). It seems a bit hidden to me. Maybe, move it to the main readme or move it to the top level?

**Comment by SebastianZimmeck** [comment]
Along those same lines, there is a [comment with future cookie functionality](https://github.com/privacy-tech-lab/gpc-optmeowt/blob/main/src/background/cookiesIAB.js#L22) that looks obsolete and that we should discuss tomorrow.

**Comment by SebastianZimmeck** [comment]
OK, published on the Chrome Web Store! 

@kalicki1, can you update the Wiki (issue #198), and then we can close that and this issue.

**Comment by kalicki1** [comment]
I just pushed a small change related to how our dev environment is setup per our last call. 

Given the minor difficulty we had in building our production releases for both Chrome and Firefox, I went ahead and added a few scripts to our `package.json` so that `npm run build` also packages everything in the distribution folder right into ready-to-ship `zip` files. These zips get dropped into the `src/dist/packages` folder. 

In addition, I also noticed that we still had an `unsafe eval` declaration in our manifest. Since we use this for our source maps in development, I split our manifest into `manifest-dev.json` and `manifest-dist.json` in the source and told webpack how to select the correct ones for our builds. This will save us from remembering to take the `unsafe eval` out for releases.
